crates/ruff_python_index/src/indexer.rs

```diff
@@ -68,18 +72,33 @@ impl Indexer {
                 Tok::Newline | Tok::NonLogicalNewline => {
                     line_start = range.end();
                 }
+                Tok::FStringStart => {
+                    f_string_start_count += 1;
+                    if f_string_start_count == 1 {
+                        first_f_string_start_range = Some(*range);
+                    }
+                }
+                Tok::FStringEnd => {
+                    // This is always going to be > 0, because the lexer will only
+                    // emit the end token if there was a start token to begin with.
+                    f_string_start_count -= 1;
+                    if f_string_start_count == 0 {
+                        if let Some(start_range) = first_f_string_start_range.take() {
+                            let f_string_range = TextRange::new(start_range.start(), range.end());
+                            f_string_ranges.push(f_string_range);
+
+                            if matches!(locator.slice(range), "'''" | r#"""""#) {
+                                triple_quoted_string_ranges.push(f_string_range);
+                            }
+                        }
+                    }
```

Comment:
`f_string_quote_style` is the only usage of `f_string_ranges`. It tries to find the f string quote style of the enclosing f-string. Does the current implementation support nested fstrings? Does the binary search still work with nested fstrings or how do we ensure we find the "smallest" range that fully encloses the expression. 

https://github.com/astral-sh/ruff/blob/04f2842e4f187a26ff74d7b845f3cdc6abacf765/crates/ruff/src/checkers/ast/mod.rs#L179-L195